From bbfd16b1ffbddd47a37b779347af87446e770e5d Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Thu, 13 Oct 2011 11:34:52 +0300
Subject: [PATCH 0353/1092] cbus: tahvo: no need to mask interrupts on exit

the children will be unloaded first and they
will make sure to mask their own IRQ.

While at that, also move subsys_init_call()
close to tahvo_init().

Signed-off-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: Tony Lindgren <tony@atomide.com>
---
 drivers/cbus/tahvo.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/cbus/tahvo.c b/drivers/cbus/tahvo.c
index d92a05d..ab6e7ea 100644
--- a/drivers/cbus/tahvo.c
+++ b/drivers/cbus/tahvo.c
@@ -470,8 +470,6 @@ static int __devexit tahvo_remove(struct platform_device *pdev)
 
 	irq = platform_get_irq(pdev, 0);
 
-	/* Mask all TAHVO interrupts */
-	__tahvo_write_reg(tahvo, TAHVO_REG_IMR, 0xffff);
 	free_irq(irq, 0);
 	irq_free_descs(tahvo->irq_base, MAX_TAHVO_IRQ_HANDLERS);
 	kfree(tahvo);
@@ -491,13 +489,12 @@ static int __init tahvo_init(void)
 {
 	return platform_driver_probe(&tahvo_driver, tahvo_probe);
 }
+subsys_initcall(tahvo_init);
 
 static void __exit tahvo_exit(void)
 {
 	platform_driver_unregister(&tahvo_driver);
 }
-
-subsys_initcall(tahvo_init);
 module_exit(tahvo_exit);
 
 MODULE_DESCRIPTION("Tahvo ASIC control");
-- 
1.7.11.2

